### PR TITLE
Irregular time series support

### PR DESCRIPTION
**Pull request by geofjamg**

### CODE REVIEW

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/IrregularTimeSeriesIndex.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    @Override
+    public int hashCode() {
+        return Arrays.hashCode(times);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj instanceof TimeSeriesIndex) {
```
Comment (mathbagu) [inline time-series/time-series-api/src/main/java/com/powsybl/timeseries/IrregularTimeSeriesIndex.java]: obj instanceof IrregularTimeSeriesIndex

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/IrregularTimeSeriesIndex.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    @Override
+    public Stream<Instant> stream() {
+        return Arrays.stream(times).mapToObj(Instant::ofEpochMilli);
+    }
+
+    @Override
+    public String toString() {
+        return "IrregularTimeSeriesIndex(times=" + Arrays.stream(times).mapToObj(Instant::ofEpochMilli).collect(Collectors.toList())
```
Comment (mathbagu) [inline time-series/time-series-api/src/main/java/com/powsybl/timeseries/IrregularTimeSeriesIndex.java]: Use the stream method:
```
return "IrregularTimeSeriesIndex(times=" + stream().collect(Collectors.toList()) +")";

```

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/IrregularTimeSeriesIndex.java (hunk 1 above)
Comment (geofjamg) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/IrregularTimeSeriesIndex.java]: Done

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/IrregularTimeSeriesIndex.java (hunk 2 above)
Comment (geofjamg) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/IrregularTimeSeriesIndex.java]: Done

**Pull request merged**